tanjun/clients.py

```diff
@@ -1616,9 +1616,9 @@ async def close(self, *, deregister_listeners: bool = True) -> None:
 
             self._try_unsubscribe(self._events, hikari.InteractionCreateEvent, self.on_interaction_create_event)
 
-            for event_type, listeners in self._listeners.items():
+            for event_type_, listeners in self._listeners.items():
```

Comment:
I don't really like adding unnecessary variable clutter like this tbh, couldn't allow redefinition just be set to true in the config to better match pyright (and also cases where it's possibly incorrectly inferring what should be narrowing the type of a variable as a redefinition)?

Reply:
I kinda forgot `--allow-redefinition` was a thing, heh

Reply:
Somewhat ironically, while allowing redefinition helped in other spots, it didn't help here.